pkg/controller/atlasproject/backupcompliancepolicy.go

```diff
@@ -0,0 +1,218 @@
+				return b.terminate(workflow.ProjectBackupCompliancePolicyNotMet, err)
+			}
+			return b.terminate(workflow.ProjectBackupCompliancePolicyNotCreatedInAtlas, err)
+		}
+	}
+
+	if atlasBCP.GetState() != "ACTIVE" {
+		return b.progress(
+			workflow.ProjectBackupCompliancePolicyUpdating,
+			fmt.Sprintf("backup compliance policy not ready yet: %q", atlasBCP.GetState()),
+			"updating backup compliance policy",
+		)
+	}
+
+	return b.idle()
+}
+
+// delete begins the deletion process for a backup compliance policy. However, the admin API cannot delete BCPs (which must be
+// done via support), so we notify the user of next steps via status.
+func (b *backupComplianceController) delete() workflow.Result {
+	b.ctx.Log.Debug("deleting backup compliance policy")
+	return b.terminate(
+		workflow.ProjectBackupCompliancePolicyCannotDelete,
+		errors.New("deletion of backup compliance policies is not supported in Atlas: please contact support"),
+	).WithoutRetry()
+}
+
+func (b *backupComplianceController) progress(state workflow.ConditionReason, fineMsg, coarseMsg string) workflow.Result {
+	var (
+		fineProgress   = workflow.InProgress(state, fineMsg)
+		coarseProgress = workflow.InProgress(state, coarseMsg)
+	)
+
+	b.ctx.SetConditionFromResult(api.BackupComplianceReadyType, fineProgress)
+	return coarseProgress
+}
+
+// terminate transitions to pending state if an error occurred.
+func (b *backupComplianceController) terminate(reason workflow.ConditionReason, err error) workflow.Result {
+	b.ctx.Log.Error(err)
+	result := workflow.Terminate(reason, err.Error())
+	b.ctx.SetConditionFromResult(api.BackupComplianceReadyType, result)
+	return result
+}
+
+// unmanage transitions to pending state if there is no managed BCP.
+func (b *backupComplianceController) unmanage() workflow.Result {
+	b.ctx.UnsetCondition(api.BackupComplianceReadyType)
+	return workflow.OK()
+}
+
+// idle transitions BCP to idle state when ready and idle.
+func (b *backupComplianceController) idle() workflow.Result {
+	b.ctx.SetConditionTrue(api.BackupComplianceReadyType)
+	return workflow.OK()
+}
+
+func (b *backupComplianceController) getAtlasBackupCompliancePolicy() (*admin.DataProtectionSettings20231001, bool, error) {
+	bcp, _, err := b.ctx.SdkClient.CloudBackupsApi.GetDataProtectionSettings(b.ctx.Context, b.project.ID()).Execute()
+	if err != nil {
+		// NOTE: getting backup compliance policies never yields a 404
+		return nil, false, fmt.Errorf("error finding backup compliance policy: %w", err)
+	}
+
+	var emptyPolicy admin.DataProtectionSettings20231001
+	if bcp == nil || reflect.DeepEqual(&emptyPolicy, bcp) {
+		return nil, false, nil
+	}
+
+	return bcp, true, nil
+}
+
+func (b *backupComplianceController) getAKOBackupCompliancePolicy() (*akov2.AtlasBackupCompliancePolicy, error) {
+	bcp := &akov2.AtlasBackupCompliancePolicy{}
+	err := b.client.Get(b.ctx.Context, *b.project.Spec.BackupCompliancePolicyRef.GetObject(b.project.Namespace), bcp)
```

Comment:
Unless you have the guarantee the ref is not nil before the code reach this call, better to check for nil pointer here

Reply:
Addressed & added a test for this